src/cli/subcommands/archive_cmd.rs

```diff
@@ -48,19 +58,128 @@ pub enum ArchiveCommands {
+        output_path: PathBuf,
+        /// Latest epoch that has to be exported for this snapshot, the upper bound. This value
+        /// cannot be greater than the latest epoch available in the input snapshot.
+        #[arg(short)]
+        epoch: ChainEpoch,
+        /// How far back we want to go. Think of it as `$epoch - $depth`, the lower bound of this
+        /// snapshot. This value cannot be less than `chain finality`, which is currently assumed
+        /// to be `900`. If this ever changes - the actual value is specified in the error message
+        /// that is thrown in case `depth` value is too low.
+        /// This parameter is optional due to the fact that we need to fetch the exact default
+        /// dynamically from a config.
+        // TODO: Investigate if we can have a dynamic default here somehow.
+        #[arg(short)]
+        depth: Option<ChainEpoch>,
+    },
 }
 
 impl ArchiveCommands {
-    pub async fn run(self, _config: Config) -> anyhow::Result<()> {
+    pub async fn run(self, config: Config) -> anyhow::Result<()> {
         match self {
             Self::Info { snapshot } => {
                 println!("{}", ArchiveInfo::from_file(snapshot)?);
                 Ok(())
             }
+            Self::Export {
+                input_path,
+                output_path,
+                epoch,
+                depth,
+            } => {
+                let chain_finality = config.chain.policy.chain_finality;
+                let depth = depth.unwrap_or(chain_finality);
+                if depth < chain_finality {
+                    bail!("depth has to be at least {}", chain_finality);
+                }
+
+                do_export(config, input_path, output_path, epoch, depth).await
+            }
         }
     }
 }
 
+async fn do_export(
+    config: Config,
+    input_path: PathBuf,
+    output_path: PathBuf,
+    epoch: ChainEpoch,
+    depth: ChainEpoch,
+) -> anyhow::Result<()> {
+    info!(
+        "indexing a car-backed store using snapshot: {}",
+        input_path.to_str().unwrap_or_default()
+    );
+
+    let store = Arc::new(
+        CarBackedBlockstore::new(std::fs::File::open(input_path)?)
+            .context("couldn't read input CAR file - it's either compressed or corrupt")?,
+    );
+
+    let genesis = read_genesis_header(
+        config.client.genesis_file.as_ref(),
+        config.chain.genesis_bytes(),
+        &store,
+    )
+    .await?;
+
+    let tmp_chain_dir = TempDir::new()?;
+
+    let chain_store = Arc::new(ChainStore::new(
+        store,
+        config.chain.clone(),
+        &genesis,
+        tmp_chain_dir.path(),
+    )?);
+
+    // TODO: This is totally unnecessary. It should be possible to do `tipset_by_height` without this step.
+    // One solution to this is making `ts` an `Option` in `tipset_by_height` method.
+    let ts = chain_store.tipset_from_keys(&TipsetKeys::new(chain_store.db.roots()))?;
```

Comment:
I added `Tipset::load` in my most recent PR to solve this exact problem. You're on your own with `tipset_by_height`, though.

```suggestion
    let ts = Tipset::load(&store, &TipsetKeys::new(chain_store.db.roots()))?;
```

Reply:
I'm not sure how this is different though, we still need to call an extra method.

Reply:
> I added `Tipset::load` in my most recent PR to solve this exact problem. You're on your own with `tipset_by_height`, though.

Does not seem to be helpful, as it returns an option and I need an `Arc`. I think I should probably revert this commit.

Reply:
You'll need it to get rid of `ChainStore`. But that refactoring can wait until a follow-up PR.